content/docs/examples/advanced-egress/egress-gateway/index.md

```diff
@@ -689,7 +689,7 @@ $ kubectl delete destinationrule egressgateway-for-cnn
 
-Also note that Istio itself *cannot securely enforce* that all the egress traffic will actually flow through the egress gateways, Istio only *enables* such flow by its sidecar proxies. If a malicious application would attack the sidecar proxy attached to the application's pod, it could bypass the sidecar proxy. Having bypassed the sidecar proxy, the malicious application could try to exit the service mesh bypassing the egress gateway, to escape the control and monitoring by Istio. It is up to the cluster administrator or the cloud provider to enforce that no traffic leaves the mesh bypassing the egress gateway. Such enforcement must be performed by mechanisms external to Istio. For example, a firewall can deny all the traffic whose source is not the egress gateway. [Kubernetes network policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) can also forbid all the egress traffic that does not originate in the egress gateway. Another possible security measure involves configuring the network in such a way that the application nodes are unable to access the Internet without directing the egress traffic through the gateway where it will be monitored and controlled. One example of such network configuration is allocating public IPs exclusively to the gateways.
+Also note that Istio itself *cannot securely enforce* that all the egress traffic will actually flow through the egress gateways, Istio only *enables* such flow by its sidecar proxies. If a malicious application would attack the sidecar proxy attached to the application's pod, it could bypass the sidecar proxy. Having bypassed the sidecar proxy, the malicious application could try to exit the service mesh bypassing the egress gateway, to escape the control and monitoring by Istio. It is up to the cluster administrator or the cloud provider to enforce that no traffic leaves the mesh bypassing the egress gateway. Such enforcement must be performed by mechanisms external to Istio. For example, a firewall can deny all the traffic whose source is not the egress gateway. [Kubernetes network policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) can also forbid all the egress traffic that does not originate in the egress gateway. Another possible security measure involves configuring the network in such a way that the application nodes are unable to access the Internet without directing the egress traffic through the gateway where it will be monitored and controlled. One example of such network configuration is allocating public IPs exclusively to the gateways and configuring [NAT](https://en.wikipedia.org/wiki/Network_address_translation) devices to drop packets that do not originate at the egress gateways.
```

Comment:
why do you need to "allocate public IPs exclusively to the gateway" AND "configure NAT devices ...". Wouldn't configuring NAT devices to drop packets not from egress gateway be sufficient?

Reply:
@frankbu Would the following sentence be more clear:

_One example of such network configuration is preventing allocation of public IPs to any pod other that the gateways, and configuring NAT devices to drop packets that do not originate at the egress gateways._

Reply:
The text has several grammar issues that reduce its clarity. I suggest the following edits:
Istio *cannot securely enforce* that all the egress traffic actually flows through the egress gateways. Istio only *enables* said flow through its sidecar proxies. If a malicious application attacks the sidecar proxy attached to the application's pod, the malicious application could bypass the sidecar proxy. Having bypassed the sidecar proxy, the malicious application could try to exit the service mesh and bypass the egress gateway. Thus, the malicious application escapes Istio's control and monitoring. The cluster administrator or the cloud provider must ensure that no traffic leaves the mesh bypassing the egress gateway. Mechanisms external to Istio must enforce this requirement. For example, a firewall can deny all traffic not coming from the egress gateway. The [Kubernetes network policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) can also forbid all the egress traffic not originating from the egress gateway. Alternatively, the cluster administrator or the cloud provider can configure  the network so, that application nodes can't access the Internet without directing the egress traffic through the gateway. The gateway can then monitor and control the egress traffic. For example, allocating public IPs exclusively to the gateways and configuring [NAT](https://en.wikipedia.org/wiki/Network_address_translation) devices to drop packets that do not originate at the egress gateways.

Reply:
Thanks, @rcaballeromx ! @frankbu What do you think? Is the paragraph at 
https://github.com/istio/istio.github.io/pull/2275#discussion_r210989118 clear enough?

Reply:
+1 on @rcaballeromx's version, up to the "Alternatively, ..." part. The rest of the rewrite has changed the meaning. Instead of:

Alternatively, the cluster administrator or the cloud provider can configure the network so, that application nodes can't access the Internet without directing the egress traffic through the gateway. The gateway can then monitor and control the egress traffic.

I suggest:

In addition, the cluster administrator or the cloud provider can configure the network so that application nodes can't access the Internet other than via a gateway. One way to do this involves preventing allocation of public IPs to pods other than gateways, and configuring NAT devices to drop packets that do not originate at the egress gateways.

Reply:
@frankbu I would alter your suggestion slightly for clarity and simplicity:
Additionally, the cluster administrator or the cloud provider can configure the network to ensure application nodes can only access the Internet via a gateway. To do this, the cluster administrator or the cloud provider can prevent the allocation of public IPs to pods other than gateways and can configure NAT devices to drop packets not originating at the egress gateways.
Note my use of "can" instead of "must" to imply that this is one possible path only, there might be others. I've simplified the sentences by removing several constructs with little to none semantic value.

Reply:
Sounds good. The whole thing is now:

Istio cannot securely enforce that all the egress traffic actually flows through the egress gateways. Istio only enables said flow through its sidecar proxies. If a malicious application attacks the sidecar proxy attached to the application's pod, the malicious application could bypass the sidecar proxy. Having bypassed the sidecar proxy, the malicious application could try to exit the service mesh and bypass the egress gateway. Thus, the malicious application escapes Istio's control and monitoring. The cluster administrator or the cloud provider must ensure that no traffic leaves the mesh bypassing the egress gateway. Mechanisms external to Istio must enforce this requirement. For example, a firewall can deny all traffic not coming from the egress gateway. The Kubernetes network policies can also forbid all the egress traffic not originating from the egress gateway. Additionally, the cluster administrator or the cloud provider can configure the network to ensure application nodes can only access the Internet via a gateway. To do this, the cluster administrator or the cloud provider can prevent the allocation of public IPs to pods other than gateways and can configure NAT devices to drop packets not originating at the egress gateways.